hydra/hydra.py

```diff
@@ -53,20 +53,38 @@ def __get_api(self, endpoint, parameters=None, headers=None):
         # r = s.send(prepared)
 
         # NOTE: Comment this out; if you use the block above.
-        r = requests.get(
-            "{}/{}".format(self.base_api_uri, endpoint),
-            params=parameters,
-            headers=headers,
-            auth=authentication,
-            verify=self.ca_certificate_path,
-        )
-        if r.status_code == 204:
-            return []
-        if r.status_code != 200:
+        try:
+            r = requests.get(
+                "{}/{}".format(self.base_api_uri, endpoint),
+                params=parameters,
+                headers=headers,
+                auth=authentication,
+                verify=self.ca_certificate_path,
+            )
+            if r.status_code == 204:
+                return []
+            if r.status_code != 200:
+                raise Exception(
+                    """looking up infomation from: {}\n
+                        Error Code: {}""".format(
+                        endpoint, r.status_code
+                    )
+                )
+            try:
+                return r.json()
+            except:
+                response = []
+                raise Exception(
+                    """looking up information from: {}\n
+                    Error returning json""".format(
+                        endpoint
+                    )
+                )
+        except Exception:
             raise Exception(
-                """looking up infomation from: {}\n
-                    Error Code: {}""".format(
-                    endpoint, r.status_code
+                """looking up information from: {}\n
+                Error during request call""".format(
+                    endpoint
                 )
             )
```

Comment:
Is there nothing from the original exception that could be used here? For example, what if the network layer fails and the call doesn't get far enough to get `r.status_code`?

Reply:
@davemulford I'm not sure I understand your question/issue here. This is the outermost exception, this is the output that drops if the network layer Does fail, therefore there will be no `r.statuscode`.

The exceptions above are if the network layer completes properly but just not with what we were hoping for. Does that help? Should I comment that in the code to help explain or am I missing the point?